packages/runtime/src/components/builtin/Text/TextV2.tsx

```diff
@@ -0,0 +1,20 @@
+import { cx } from '@emotion/css'
+import { ElementIDValue } from '../../../prop-controllers/descriptors'
+import { RichTextV2ControlValue } from '../../../runtimes/react/controls/rich-text-v2'
+
+type Props = {
+  id?: ElementIDValue
+  text?: RichTextV2ControlValue
+  width?: string
+  margin?: string
+}
+
+function Text({ id, text, width, margin }: Props) {
```

Comment:
I think we should rename the `text` prop to `children`. Also, instead of taking `width` and `margin`, we should take `className` and use our `Style` control.

But perhaps more importantly, which would invalidate the above, how come we're creating a new component? I would've expect we just use the new control with the existing component

Reply:
> instead of taking width and margin, we should take className and use our Style control.
I can't do this without creating a migration from Style prop controller to Style control. 

> how come we're creating a new component?
The old component took rich text data / used the RichText prop controller. This new component needs to expect a react node.

Reply:
Ok, I think we should either update the existing component to handle the new control or create a brand new component. If we keep the existing component, it's a matter of updating the registration to use the new control and then the implementation of the component so that it treats the `text` prop as a `ReactNode`. If we do a new component, I think it should have a separate type and given that it's a new component I would apply the latest patterns and avoid legacy ones (i.e., use Style control and better prop names). Also, if we do a new component we'd need to hide the other component and also make sure that the text shortcut is using the new component, which will be a bit more involved than just updating the existing component.

So, I would suggest you just update the existing component. And it really isn't a different component since it has the same type when its registered and the same props, etc.

Reply:
I was hesitant to delete the old code for the Text component, so I "upgraded" it by creating this component and changing the registration of the `Text` component. 

I can instead actually update the `Text.tsx` file so that it's all in version history